modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/acls/AclIndexer.scala

```diff
@@ -27,26 +40,62 @@ class AclIndexer[F[_]](client: SparqlClient[F], settings: AclIndexingSettings)(i
     * @return a Unit value in the ''F[_]'' context
     */
   final def apply(event: Event): F[Unit] = event match {
-    case PermissionsCreated(path, acl, meta) =>
-      acl.acl.map { case AccessControl(id, perms) if perms(Permission.Read) => add(path, id) }
-      F.pure(())
-    case PermissionsAdded(path, identity, permissions, meta) if permissions(Permission.Read) =>
-      add(path, identity)
-    case PermissionsSubtracted(path, identity, permissions, meta) if permissions(Permission.Read) =>
+    case PermissionsCreated(path, acl, _) =>
+      val identities = acl.acl.filter(_.permissions(Permission.Read)).map(_.identity)
+      add(path, identities)
+    case PermissionsAdded(path, identity, permissions, _) if permissions(Permission.Read) && isKg(path) =>
+      add(path, Set(identity))
+    case PermissionsSubtracted(path, identity, permissions, _) if permissions(Permission.Read) && isKg(path) =>
       remove(path, identity)
-    case PermissionsCleared(path, meta) =>
+    case PermissionsCleared(path, _) if isKg(path) =>
       clear(path)
-    case PermissionsRemoved(path, identity, meta) =>
+    case PermissionsRemoved(path, identity, _) if isKg(path) =>
       remove(path, identity)
     case _ => F.pure(())
   }
 
-  // TODO: implement
-  private def add(path: Path, identity: Identity): F[Unit] = ???
+  private def isKg(path: Path): Boolean =
+    Path(s"${path.reverse.head}") == Path("kg")
```

Comment:
Wouldn't `path.reverse.head.toString == "kg"` work the same?

Reply:
If the path in the `Event` is just `/kg` then it would be `"/kg" == "kg"`
So it is to double check special cases with slashes